src/NHibernate/Loader/Loader.cs

```diff
@@ -944,25 +944,68 @@ private void CheckVersion(int i, IEntityPersister persister, object id, object e
 
 			object[] rowResults = new object[cols];
 
+			var entityCounter = new Dictionary<string, int>();
+
 			for (int i = 0; i < cols; i++)
 			{
 				object obj = null;
+				ILoadable persister = persisters[i];
+				
+				if (entityCounter.ContainsKey(persister.EntityName))
+				{
+					entityCounter[persister.EntityName]++;
+				}
+				else
+				{
+					entityCounter.Add(persister.EntityName, 1);
+				}
+
+				int total = entityCounter[persister.EntityName];
+
 				EntityKey key = keys[i];
 
 				if (key == null)
 				{
-					// do nothing
-					/* TODO NH-1001 : if (persisters[i]...EntityType) is an OneToMany or a ManyToOne and
-					 * the keys.length > 1 and the relation IsIgnoreNotFound probably we are in presence of
-					 * an load with "outer join" the relation can be considerer loaded even if the key is null (mean not found)
-					*/
+					if (cols > 1)
+					{
+						/*
+						* the keys.length > 1 and the relation IsIgnoreNotFound probably we are in presence of
```

Comment:
In original comment is also mentioned `OneToMany type. But I see only changes related to ManyToOne. OneToMany case still not fixed?

Reply:
I will add test for this case.